enumerable_methods.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_each { |e| yield(e, index(e)) }
+    self
+  end
```

Comment:
In `my_each_with_index`
- my_each_with_index calls the given block once for each element in self